pkg/deploy/dev-workspace/dev_workspace.go

```diff
@@ -148,23 +155,38 @@ func ReconcileDevWorkspace(deployContext *deploy.DeployContext) (bool, error) {
 		return true, nil
 	}
 
-	// check if DW exists on the cluster
-	devWorkspaceWebhookExists, err := deploy.Get(
-		deployContext,
-		client.ObjectKey{Name: DevWorkspaceWebhookName},
-		&admissionregistrationv1.MutatingWebhookConfiguration{},
-	)
+	isCreated, err := createDwNamespace(deployContext)
 	if err != nil {
 		return false, err
 	}
+	if !isCreated {
+		namespace := &corev1.Namespace{}
+		err := deployContext.ClusterAPI.NonCachedClient.Get(context.TODO(), types.NamespacedName{Name: DevWorkspaceNamespace}, namespace)
+		if err != nil {
+			return false, err
+		}
 
-	if devWorkspaceWebhookExists {
-		// if DW exists then check if version matches
-		if err := checkWebTerminalSubscription(deployContext); err != nil {
+		namespaceOwnershipAnnotation := namespace.GetAnnotations()[deploy.CheEclipseOrgNamespace]
```

Comment:
check 'namespace.GetAnnotations() != nil'

Reply:
why? As far as I can tell, it's safe to use nil map for read. I can double check.